modules/imaging_browser/php/imaging_browser.class.inc

```diff
@@ -284,117 +98,119 @@ END";
-     *         Overloading this method to create a list of sessionID that
-     *         will be used for the Navigation Links in  the viewSession
-     *         page.
+     * @return \LORIS\Data\Provisioner
+     */
+    public function getBaseDataProvisioner() : \LORIS\Data\Provisioner
+    {
+        return new ImagingBrowserRowProvisioner();
+    }
+
+    /**
+     * Return a data provisioner of the same type as BaseDataProvisioner, with
+     * additional LORIS filters applied.
      *
-     * @return array
+     * @return \LORIS\Data\Provisioner a provisioner with specific filters added
      */
-    function toArray()
+    public function getDataProvisionerWithFilters(): \LORIS\Data\Provisioner
     {
-        $data  = parent::toArray();
-        $index = array_search('SessionID', $data['Headers']);
-        if ($index !== false) {
-            $_SESSION['State']->setProperty(
-                'mriSessionsListed',
-                array_column($data['Data'], $index)
-            );
+
+        $provisioner = parent::getDataProvisionerWithFilters();
+
+        $user = \User::singleton();
+
+        // UserSiteMatch for imaging browser specific permissions
+        if ($user->hasPermission("imaging_browser_view_allsites") == false) {
+            if ($user->hasPermission("imaging_browser_view_site") == false) {
+                // user only has access to phantom scans from all sites
+                $provisioner = $provisioner->filter(
+                    new \LORIS\Data\Filters\OnlyPhantoms()
+                );
+                if ($user->hasPermission("imaging_browser_phantom_allsites") == false) {
+                    // user only has access to own site's phantom scans
+                    $provisioner = $provisioner->filter(
+                        new \LORIS\Data\Filters\UserSiteMatch()
+                    );
+                }
+            } else {
+                if ($user->hasPermission('imaging_browser_phantom_allsites')) {
+                    // user has access to own site scans and all sites phantom
+                    // TODO
+                    $provisioner = $provisioner->filter(
+                        new \LORIS\Data\Filters\AllSitesPhantomsOwnSitesScans()
+                    );
+                } else if ($user->hasPermission('imaging_browser_phantom_ownsite')) {
+                    // user has access to own site scans and own site phantom
+                    $provisioner = $provisioner->filter(
+                        new \LORIS\Data\Filters\UserSiteMatch()
+                    );
+                } else {
+                    // user has access to own site scans only
+                    $provisioner = $provisioner->filter(
+                        new \LORIS\Data\Filters\UserSiteMatch()
+                    );
+                    $provisioner = $provisioner->filter(
+                        new \LORIS\Data\Filters\NoPhantoms()
+                    );
+                }
+            }
```

Comment:
missing else statement for when the user do have `imaging_browser_view_allsites` but might need to filter out phantoms.

Reply:
indeed. Will add it shortly!